OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
My confusion really comes down to it not being clear what `NextActivity` can/will return in different cases.  I hope my other comments gives some direction on how to fix this, even if its just adding some comments for now, to be fixed more deeply (ideally with a clear split between child and next activities) in future steps.